### PR TITLE
bug-fix-addTest

### PR DESCRIPTION
**Pull request by DhruvCopods**
Fixed the following issue:
- In Add Test, Step 2 and Step 3 were accessible while the name and description fields were empty, those clickable steps were creating UI flickering

### CODE REVIEW

File: app/components/tests/AddTest.tsx [hunk 1]
```diff
@@ -125,6 +125,8 @@ const AddTestComponent = ({ sections }: { sections: Array<TestSection> }) => {
       </div>
       <StepsTabComponent
         tabs={tabs}
+        name={name}
```
Comment (richa-copods) [inline app/components/tests/AddTest.tsx:128]: Instead of sending `name` and `description` we can directly pass a boolean prop called isActive

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/tests/StepsTab.tsx [hunk 2]
```diff
@@ -12,12 +16,12 @@ const StepsTabComponent = ({
...
-            id={tab.id.toString()}
-            className="flex-1 cursor-pointer"
-            onClick={() => {
-              setCurrentTab(tab.id)
-            }}
+          key={tab.id}
+          id={tab.id.toString()}
+          className={`flex-1 ${name === '' || description === '' && currentTab !== tab.id ? 'cursor-not-allowed' : 'cursor-pointer'}`} 
+          onClick={() => {
+            name !== '' && description !== '' && setCurrentTab(tab.id)
```
[the remark this answers is not in the document]
Comment (DhruvCopods) [reply app/components/tests/StepsTab.tsx:23]: Changed the class to pointer-events-none, thanks for pointing it out

File: app/components/tests/AddTest.tsx (hunk 1 above)
Comment (DhruvCopods) [reply app/components/tests/AddTest.tsx:128]: Thanks for pointing it out, created isEmpty boolean prop

File: app/components/tests/StepsTab.tsx [hunk 3]
```diff
@@ -12,12 +16,12 @@ const StepsTabComponent = ({
...
           <div
-            key={tab.id}
-            id={tab.id.toString()}
-            className="flex-1 cursor-pointer"
-            onClick={() => {
-              setCurrentTab(tab.id)
-            }}
+          key={tab.id}
+          id={tab.id.toString()}
+          className={`flex-1 ${name === '' || description === '' && currentTab !== tab.id ? 'cursor-not-allowed' : 'cursor-pointer'}`}
```
[the remark this answers is not in the document]
Comment (DhruvCopods) [reply app/components/tests/StepsTab.tsx]: Created a isEmpty boolean prop, also while rechecking the code the currentTab condition was not necessary so I have removed that.

File: app/components/tests/StepsTab.tsx [hunk 4]
```diff
@@ -12,12 +14,12 @@ const StepsTabComponent = ({
...
-            id={tab.id.toString()}
-            className="flex-1 cursor-pointer"
-            onClick={() => {
-              setCurrentTab(tab.id)
-            }}
+          key={tab.id}
+          id={tab.id.toString()}
+          className={`flex-1 ${isDisabled ? 'pointer-events-none' : 'cursor-pointer'}`} 
+          onClick={() => {
+            setCurrentTab(tab.id)
```
Comment (anuragP04) [inline app/components/tests/StepsTab.tsx:21]: You are only disabling click using css, but ideally setCurrentTab function should also be blocked/disabled to avoid glitch in case I change CSS property.

Comment (DhruvCopods) [reply app/components/tests/StepsTab.tsx]: Added conditional calling to setCurrentTab

File: app/components/tests/StepsTab.tsx [hunk 5]
```diff
@@ -12,12 +14,12 @@ const StepsTabComponent = ({
...
-            id={tab.id.toString()}
-            className="flex-1 cursor-pointer"
-            onClick={() => {
-              setCurrentTab(tab.id)
-            }}
+          key={tab.id}
+          id={tab.id.toString()}
+          className={`flex-1 ${isDisabled ? 'pointer-events-none' : 'cursor-pointer'}`} 
+          onClick={() => {
+            !isDisabled && setCurrentTab(tab.id)
```
Comment (arshabh-copods) [inline app/components/tests/StepsTab.tsx:21]: with pointer-events-none - this is not needed

**Review by arshabh-copods — CHANGES_REQUESTED** [review]

File: app/components/tests/StepsTab.tsx (hunk 5 above)
Comment (DhruvCopods) [reply app/components/tests/StepsTab.tsx]: removed the condition

**Review by arshabh-copods — APPROVED** [review]

**Review by anuragP04 — APPROVED** [review]

**Review by richa-copods — APPROVED** [review]
